generic: 6.6: mtk_eth_soc: reset all TX queues on DMA free

dangowrt opened the pull request:
The purpose of resetting the TX queue is to reset the byte and packet count as well as to clear the software flow control XOFF bit.

MediaTek developers pointed out that netdev_reset_queue would only resets queue 0 of the network device.
Queues that are not reset may cause unexpected issues.

Packets may stop being sent after reset and "transmit timeout" log may be displayed.

Import fix from MediaTek's SDK to resolve this issue.

Link: https://git01.mediatek.com/plugins/gitiles/openwrt/feeds/mtk-openwrt-feeds/+/5746a94456f466446cc0dcdfbd9078df6df31b63

May fix https://github.com/openwrt/openwrt/issues/13122

dangowrt committed:
generic: 6.6: mtk_eth_soc: reset all TX queues on DMA free

The purpose of resetting the TX queue is to reset the
byte and packet count as well as to clear the software
flow control XOFF bit.

MediaTek developers pointed out that netdev_reset_queue would only
resets queue 0 of the network device.
Queues that are not reset may cause unexpected issues.

Packets may stop being sent after reset and "transmit timeout" log may
be displayed.

Import fix from MediaTek's SDK to resolve this issue.

Link: https://git01.mediatek.com/plugins/gitiles/openwrt/feeds/mtk-openwrt-feeds/+/5746a94456f466446cc0dcdfbd9078df6df31b63
Signed-off-by: Daniel Golle <daniel@makrotopia.org>

neheb commented:
sounds applicable to the local driver as well.

cmonroe commented:
> Also for netsys v2, pointed by cmonroe [cmonroe/target-mediatek@e5062a8](https://github.com/cmonroe/target-mediatek/commit/e5062a8a84f62e79965d009afd9759a18224077f)

I was just coming here to post this, thanks. We've been working thru some intermittent tx timeout issues and this patch was one of the recommendations. I don't have any data on whether it helps yet or not but should over the next few weeks.

dangowrt commented:
>And do we actually need [this](https://github.com/openwrt/openwrt/blob/d97dced25945bd1c4c4653eefd85cb2d05f4ad0c/target/linux/generic/pending-6.6/705-net-dsa-tag_mtk-add-padding-for-tx-packets.patch)?

I think this patch is a more elegant solution to address the same problem as the two commits in SDK:

> [][Kernel][mt7988][eth][Add sw padding in Ethernet Tx] [frank-w/BPI-Router-Linux@636acc2](https://github.com/frank-w/BPI-Router-Linux/commit/636acc25e2961a67b9fe318c948c00447f5047e4) and [][HIGH][kernel][mt7988][eth][Fix software padding processing in Ethernet Tx path] [frank-w/BPI-Router-Linux@395a8cd](https://github.com/frank-w/BPI-Router-Linux/commit/395a8cd3bb62cff868bbce7dc8631b82daead7f5)

Reading the description of those it's also clear that the issue is the DSA switch requiring a minimum length of packets with MTK special tag which is used on the CPU port. @nbd168 patch above solves this in a nicer way without adding padding in case the MAC is directly driving a PHY without any built-in or external MT753x DSA switch.

RomanovDanila commented:
> without adding padding in case the MAC is directly driving a PHY without any built-in or external MT753x DSA switch

Perhaps that's what we need, since it helped. @axet 
>With this patch router **no longer crashing**. But showing timeout as dmesg error

axet commented:
Are you properly back-porting the patch? Original saying `txqs = MTK_PDMA_TX_NUM` which should be translated to `txqs = 4`. But your code saying `txqs = 1`.

* https://github.com/GainStrongService/mtk-openwrt-feeds/commit/5746a94456f466446cc0dcdfbd9078df6df31b63
* https://github.com/GainStrongService/mtk-openwrt-feeds/blob/7b4fa7089f1d09b37ca1456e7ead28cff408ff42/21.02/files/target/linux/mediatek/files-5.4/drivers/net/ethernet/mediatek/mtk_eth_soc.h#L692

dangowrt commented:
> Original saying `txqs = MTK_PDMA_TX_NUM` which should be translated to `txqs = 4`. But your code saying `txqs = 1`.

In the non-QDMA case the vanilla driver is using only 1 queue, see
https://github.com/torvalds/linux/blob/master/drivers/net/ethernet/mediatek/mtk_eth_soc.c#L4542

dangowrt commented:
> Perhaps that's what we need, since it helped. @axet
> 
> > With this patch router **no longer crashing**. But showing timeout as dmesg error

Note that this was referring to the patch applied on top of @frank-w BPI-Router-Linux Linux repo, not OpenWrt which got the alternative patch.

dangowrt committed:
generic: 6.6: mtk_eth_soc: increase QDMA RESV_BUF size

Increase QDMA RESV_BUF from 2K to 3K for netsys v2 to match Mediatek SDK.
This helps reduce the possibility of Ethernet transmit timeouts.

Link: https://git01.mediatek.com/plugins/gitiles/openwrt/feeds/mtk-openwrt-feeds/+/19d8456c3051e5f6dabf42fa770916a2126ea4bf
Signed-off-by: Daniel Golle <daniel@makrotopia.org>

DragonBluep commented:
> > Also for netsys v2, pointed by cmonroe [cmonroe/target-mediatek@e5062a8](https://github.com/cmonroe/target-mediatek/commit/e5062a8a84f62e79965d009afd9759a18224077f)
> 
> I was just coming here to post this, thanks. We've been working thru some intermittent tx timeout issues and this patch was one of the recommendations. I don't have any data on whether it helps yet or not but should over the next few weeks.

The timeout issue should be caused by some common code. I remember some MT7621 users already complained it several years ago, before MT798x released.

For this PR, LGTM.

hackpascal commented:
I've tried this earlier, and it solved the issue on my bpi-r4:
https://pastebin.com/EAhMni4v

frank-w commented:
> I've tried this earlier, and it solved the issue on my bpi-r4: https://pastebin.com/EAhMni4v

Have you done reset via register writes? I ask because i see the timeouts in your log.

Imho the reset patch only fixes symptoms (and prevent the annoying watchdog which freezes system) but not underlaying issue where the tx packets are not sent,or am i wrong?

hackpascal commented:
> Imho the reset patch only fixes symptoms (and prevent the annoying watchdog which freezes system) but not underlaying issue where the tx packets are not sent,or am i wrong?

yeah.. the root issue was not fixed. but it can save my router without reboot

frank-w commented:
Have you also done the rcvbuf patch and the [padding in tag-dsa driver](https://github.com/openwrt/openwrt/blob/d97dced25945bd1c4c4653eefd85cb2d05f4ad0c/target/linux/generic/pending-6.6/705-net-dsa-tag_mtk-add-padding-for-tx-packets.patch)? They could maybe fix the rootcause.

The rcvbuf change was also suggested by mtk in my conversation,i guess this is because it is in this PR :)

hackpascal commented:
> Have you also done the rcvbuf patch and the [padding in tag-dsa driver](https://github.com/openwrt/openwrt/blob/d97dced25945bd1c4c4653eefd85cb2d05f4ad0c/target/linux/generic/pending-6.6/705-net-dsa-tag_mtk-add-padding-for-tx-packets.patch)? They could maybe fix the rootcause.

I'm using vanilla openwrt main branch right now. Of course this patch is included.

NovaPixell commented:
This patch may also fix #13122?

axet commented:
This is too early too tell. My router used to work for 3 months before issue occured first time. We need at least 6-12 months to test. Before confirm it is a solution. I doubt.

sunnyqeen commented:
@dangowrt could you please backport this pr to 23.05

axet commented:
Router crashing and stop to respond (same as before) with different dmesg:

```
[67139.043056] mt7530-mdio mdio-bus:1f lan4: Link is Up - 100Mbps/Full - flow control rx/tx
[67139.044202] br-lan: port 4(lan4) entered blocking state
[67139.069661] br-lan: port 4(lan4) entered forwarding state
[67140.578057] mt7530-mdio mdio-bus:1f lan4: Link is Down
[67140.588491] br-lan: port 4(lan4) entered disabled state
[67142.983047] mt7530-mdio mdio-bus:1f lan4: Link is Up - 100Mbps/Full - flow control rx/tx
[67142.999360] br-lan: port 4(lan4) entered blocking state
[67143.009840] br-lan: port 4(lan4) entered forwarding state
[115361.763487] mtk_soc_eth 1e100000.ethernet eth0: NETDEV WATCHDOG: CPU: 1: transmit queue 4 timed out 5980 ms
[115366.723427] mtk_soc_eth 1e100000.ethernet eth0: NETDEV WATCHDOG: CPU: 1: transmit queue 4 timed out 10940 ms
[115371.763366] mtk_soc_eth 1e100000.ethernet eth0: NETDEV WATCHDOG: CPU: 1: transmit queue 4 timed out 15980 ms
[115376.723317] mtk_soc_eth 1e100000.ethernet eth0: NETDEV WATCHDOG: CPU: 1: transmit queue 4 timed out 20940 ms
[115381.763250] mtk_soc_eth 1e100000.ethernet eth0: NETDEV WATCHDOG: CPU: 1: transmit queue 4 timed out 25980 ms
[115386.723194] mtk_soc_eth 1e100000.ethernet eth0: NETDEV WATCHDOG: CPU: 1: transmit queue 4 timed out 30940 ms
[115391.763127] mtk_soc_eth 1e100000.ethernet eth0: NETDEV WATCHDOG: CPU: 1: transmit queue 4 timed out 35980 ms
[115396.723074] mtk_soc_eth 1e100000.ethernet eth0: NETDEV WATCHDOG: CPU: 1: transmit queue 4 timed out 40940 ms
[115401.762993] mtk_soc_eth 1e100000.ethernet eth0: NETDEV WATCHDOG: CPU: 1: transmit queue 4 timed out 45980 ms
[115406.722946] mtk_soc_eth 1e100000.ethernet eth0: NETDEV WATCHDOG: CPU: 1: transmit queue 4 timed out 50940 ms
[115411.762882] mtk_soc_eth 1e100000.ethernet eth0: NETDEV WATCHDOG: CPU: 1: transmit queue 4 timed out 55980 ms
[115416.722821] mtk_soc_eth 1e100000.ethernet eth0: NETDEV WATCHDOG: CPU: 1: transmit queue 4 timed out 60940 ms
[115421.762759] mtk_soc_eth 1e100000.ethernet eth0: NETDEV WATCHDOG: CPU: 1: transmit queue 4 timed out 65980 ms
[115426.722703] mtk_soc_eth 1e100000.ethernet eth0: NETDEV WATCHDOG: CPU: 1: transmit queue 4 timed out 70940 ms
[115431.762626] mtk_soc_eth 1e100000.ethernet eth0: NETDEV WATCHDOG: CPU: 1: transmit queue 4 timed out 75980 ms
[115436.722591] mtk_soc_eth 1e100000.ethernet eth0: NETDEV WATCHDOG: CPU: 1: transmit queue 4 timed out 80940 ms
[115441.762516] mtk_soc_eth 1e100000.ethernet eth0: NETDEV WATCHDOG: CPU: 1: transmit queue 4 timed out 85980 ms
[115446.722454] mtk_soc_eth 1e100000.ethernet eth0: NETDEV WATCHDOG: CPU: 1: transmit queue 4 timed out 90940 ms
[115451.762403] mtk_soc_eth 1e100000.ethernet eth0: NETDEV WATCHDOG: CPU: 1: transmit queue 4 timed out 95980 ms
[115456.722338] mtk_soc_eth 1e100000.ethernet eth0: NETDEV WATCHDOG: CPU: 1: transmit queue 4 timed out 100940 ms
[115461.762270] mtk_soc_eth 1e100000.ethernet eth0: NETDEV WATCHDOG: CPU: 1: transmit queue 4 timed out 105980 ms
[115466.722219] mtk_soc_eth 1e100000.ethernet eth0: NETDEV WATCHDOG: CPU: 1: transmit queue 4 timed out 110940 ms
[115471.762151] mtk_soc_eth 1e100000.ethernet eth0: NETDEV WATCHDOG: CPU: 1: transmit queue 4 timed out 115980 ms
[115476.722099] mtk_soc_eth 1e100000.ethernet eth0: NETDEV WATCHDOG: CPU: 1: transmit queue 4 timed out 120940 ms
[115481.762036] mtk_soc_eth 1e100000.ethernet eth0: NETDEV WATCHDOG: CPU: 1: transmit queue 4 timed out 125980 ms
[115486.721971] mtk_soc_eth 1e100000.ethernet eth0: NETDEV WATCHDOG: CPU: 1: transmit queue 4 timed out 130940 ms
[115491.761899] mtk_soc_eth 1e100000.ethernet eth0: NETDEV WATCHDOG: CPU: 1: transmit queue 4 timed out 135980 ms
[115496.721853] mtk_soc_eth 1e100000.ethernet eth0: NETDEV WATCHDOG: CPU: 1: transmit queue 4 timed out 140940 ms
[115501.761786] mtk_soc_eth 1e100000.ethernet eth0: NETDEV WATCHDOG: CPU: 1: transmit queue 4 timed out 145980 ms
[115506.721712] mtk_soc_eth 1e100000.ethernet wan: NETDEV WATCHDOG: CPU: 1: transmit queue 1 timed out 8820 ms
[115506.741166] mtk_soc_eth 1e100000.ethernet eth0: NETDEV WATCHDOG: CPU: 1: transmit queue 4 timed out 150940 ms
[115511.761658] mtk_soc_eth 1e100000.ethernet eth0: NETDEV WATCHDOG: CPU: 1: transmit queue 4 timed out 155980 ms
[115511.781615] mtk_soc_eth 1e100000.ethernet wan: NETDEV WATCHDOG: CPU: 1: transmit queue 1 timed out 13860 ms
[115516.721599] mtk_soc_eth 1e100000.ethernet wan: NETDEV WATCHDOG: CPU: 1: transmit queue 1 timed out 18820 ms
[115516.741229] mtk_soc_eth 1e100000.ethernet eth0: NETDEV WATCHDOG: CPU: 1: transmit queue 4 timed out 160940 ms
[115521.761543] mtk_soc_eth 1e100000.ethernet eth0: NETDEV WATCHDOG: CPU: 1: transmit queue 4 timed out 165980 ms
[115521.781508] mtk_soc_eth 1e100000.ethernet wan: NETDEV WATCHDOG: CPU: 1: transmit queue 1 timed out 23880 ms
[115526.721488] mtk_soc_eth 1e100000.ethernet wan: NETDEV WATCHDOG: CPU: 1: transmit queue 1 timed out 28820 ms
[115526.741118] mtk_soc_eth 1e100000.ethernet eth0: NETDEV WATCHDOG: CPU: 1: transmit queue 4 timed out 170940 ms
[115531.761423] mtk_soc_eth 1e100000.ethernet eth0: NETDEV WATCHDOG: CPU: 1: transmit queue 4 timed out 175980 ms
[115531.781385] mtk_soc_eth 1e100000.ethernet wan: NETDEV WATCHDOG: CPU: 1: transmit queue 1 timed out 33870 ms
[115536.721353] mtk_soc_eth 1e100000.ethernet wan: NETDEV WATCHDOG: CPU: 1: transmit queue 1 timed out 38820 ms
[115536.740980] mtk_soc_eth 1e100000.ethernet eth0: NETDEV WATCHDOG: CPU: 1: transmit queue 4 timed out 180950 ms
[115541.761309] mtk_soc_eth 1e100000.ethernet eth0: NETDEV WATCHDOG: CPU: 1: transmit queue 4 timed out 185980 ms
[115541.781296] mtk_soc_eth 1e100000.ethernet wan: NETDEV WATCHDOG: CPU: 1: transmit queue 1 timed out 43880 ms
[115546.721236] mtk_soc_eth 1e100000.ethernet wan: NETDEV WATCHDOG: CPU: 1: transmit queue 1 timed out 48820 ms
[115546.740849] mtk_soc_eth 1e100000.ethernet eth0: NETDEV WATCHDOG: CPU: 1: transmit queue 4 timed out 190940 ms
[115551.761172] mtk_soc_eth 1e100000.ethernet eth0: NETDEV WATCHDOG: CPU: 1: transmit queue 4 timed out 195980 ms
[115551.781133] mtk_soc_eth 1e100000.ethernet wan: NETDEV WATCHDOG: CPU: 1: transmit queue 1 timed out 53870 ms
[115556.721119] mtk_soc_eth 1e100000.ethernet wan: NETDEV WATCHDOG: CPU: 1: transmit queue 1 timed out 58820 ms
[115556.740732] mtk_soc_eth 1e100000.ethernet eth0: NETDEV WATCHDOG: CPU: 1: transmit queue 4 timed out 200940 ms
[115561.761049] mtk_soc_eth 1e100000.ethernet eth0: NETDEV WATCHDOG: CPU: 1: transmit queue 4 timed out 205980 ms
[115561.781005] mtk_soc_eth 1e100000.ethernet wan: NETDEV WATCHDOG: CPU: 1: transmit queue 1 timed out 63860 ms
[115566.720988] mtk_soc_eth 1e100000.ethernet wan: NETDEV WATCHDOG: CPU: 1: transmit queue 1 timed out 68820 ms
[115566.740614] mtk_soc_eth 1e100000.ethernet eth0: NETDEV WATCHDOG: CPU: 1: transmit queue 4 timed out 210950 ms
[115571.760948] mtk_soc_eth 1e100000.ethernet eth0: NETDEV WATCHDOG: CPU: 1: transmit queue 4 timed out 215980 ms
[115571.780951] mtk_soc_eth 1e100000.ethernet wan: NETDEV WATCHDOG: CPU: 1: transmit queue 1 timed out 73880 ms
[115576.720866] mtk_soc_eth 1e100000.ethernet wan: NETDEV WATCHDOG: CPU: 1: transmit queue 1 timed out 78820 ms
[115576.740484] mtk_soc_eth 1e100000.ethernet eth0: NETDEV WATCHDOG: CPU: 1: transmit queue 4 timed out 220940 ms
[115581.760819] mtk_soc_eth 1e100000.ethernet eth0: NETDEV WATCHDOG: CPU: 1: transmit queue 4 timed out 225980 ms
[115581.780782] mtk_soc_eth 1e100000.ethernet wan: NETDEV WATCHDOG: CPU: 1: transmit queue 1 timed out 83880 ms
[115586.720752] mtk_soc_eth 1e100000.ethernet wan: NETDEV WATCHDOG: CPU: 1: transmit queue 1 timed out 88820 ms
[115586.740367] mtk_soc_eth 1e100000.ethernet eth0: NETDEV WATCHDOG: CPU: 1: transmit queue 4 timed out 230940 ms
[115591.760693] mtk_soc_eth 1e100000.ethernet eth0: NETDEV WATCHDOG: CPU: 1: transmit queue 4 timed out 235980 ms
[115591.780651] mtk_soc_eth 1e100000.ethernet wan: NETDEV WATCHDOG: CPU: 1: transmit queue 1 timed out 93860 ms
[115596.720623] mtk_soc_eth 1e100000.ethernet wan: NETDEV WATCHDOG: CPU: 1: transmit queue 1 timed out 98820 ms
[115596.740250] mtk_soc_eth 1e100000.ethernet eth0: NETDEV WATCHDOG: CPU: 1: transmit queue 4 timed out 240950 ms
[115601.760567] mtk_soc_eth 1e100000.ethernet eth0: NETDEV WATCHDOG: CPU: 1: transmit queue 4 timed out 245980 ms
[115601.780524] mtk_soc_eth 1e100000.ethernet wan: NETDEV WATCHDOG: CPU: 1: transmit queue 1 timed out 103860 ms
[115606.720518] mtk_soc_eth 1e100000.ethernet wan: NETDEV WATCHDOG: CPU: 1: transmit queue 1 timed out 108820 ms
[115606.740307] mtk_soc_eth 1e100000.ethernet eth0: NETDEV WATCHDOG: CPU: 1: transmit queue 4 timed out 250950 ms
[115611.760456] mtk_soc_eth 1e100000.ethernet eth0: NETDEV WATCHDOG: CPU: 1: transmit queue 4 timed out 255980 ms
[115611.780436] mtk_soc_eth 1e100000.ethernet wan: NETDEV WATCHDOG: CPU: 1: transmit queue 1 timed out 113860 ms
[115616.720387] mtk_soc_eth 1e100000.ethernet wan: NETDEV WATCHDOG: CPU: 1: transmit queue 1 timed out 118820 ms
[115616.740175] mtk_soc_eth 1e100000.ethernet eth0: NETDEV WATCHDOG: CPU: 1: transmit queue 4 timed out 260940 ms
[115621.760341] mtk_soc_eth 1e100000.ethernet eth0: NETDEV WATCHDOG: CPU: 1: transmit queue 4 timed out 265980 ms
[115621.780306] mtk_soc_eth 1e100000.ethernet wan: NETDEV WATCHDOG: CPU: 1: transmit queue 1 timed out 123880 ms
[115626.720261] mtk_soc_eth 1e100000.ethernet wan: NETDEV WATCHDOG: CPU: 1: transmit queue 1 timed out 128820 ms
[115626.740048] mtk_soc_eth 1e100000.ethernet eth0: NETDEV WATCHDOG: CPU: 1: transmit queue 4 timed out 270950 ms
[115631.760209] mtk_soc_eth 1e100000.ethernet eth0: NETDEV WATCHDOG: CPU: 1: transmit queue 4 timed out 275980 ms
[115631.780240] mtk_soc_eth 1e100000.ethernet wan: NETDEV WATCHDOG: CPU: 1: transmit queue 1 timed out 133880 ms
[115636.720149] mtk_soc_eth 1e100000.ethernet wan: NETDEV WATCHDOG: CPU: 1: transmit queue 1 timed out 138820 ms
[115636.739938] mtk_soc_eth 1e100000.ethernet eth0: NETDEV WATCHDOG: CPU: 1: transmit queue 4 timed out 280940 ms
[115641.760076] mtk_soc_eth 1e100000.ethernet eth0: NETDEV WATCHDOG: CPU: 1: transmit queue 4 timed out 285980 ms
[115641.780032] mtk_soc_eth 1e100000.ethernet wan: NETDEV WATCHDOG: CPU: 1: transmit queue 1 timed out 143860 ms
[115646.720038] mtk_soc_eth 1e100000.ethernet wan: NETDEV WATCHDOG: CPU: 1: transmit queue 1 timed out 148820 ms
[115646.739871] mtk_soc_eth 1e100000.ethernet eth0: NETDEV WATCHDOG: CPU: 1: transmit queue 4 timed out 290950 ms
[115651.759953] mtk_soc_eth 1e100000.ethernet eth0: NETDEV WATCHDOG: CPU: 1: transmit queue 4 timed out 295980 ms
[115651.779910] mtk_soc_eth 1e100000.ethernet wan: NETDEV WATCHDOG: CPU: 1: transmit queue 1 timed out 153860 ms
[115656.719901] mtk_soc_eth 1e100000.ethernet wan: NETDEV WATCHDOG: CPU: 1: transmit queue 1 timed out 158820 ms
[115656.739706] mtk_soc_eth 1e100000.ethernet eth0: NETDEV WATCHDOG: CPU: 1: transmit queue 4 timed out 300950 ms
[115661.759834] mtk_soc_eth 1e100000.ethernet eth0: NETDEV WATCHDOG: CPU: 1: transmit queue 4 timed out 305980 ms
[115661.779805] mtk_soc_eth 1e100000.ethernet wan: NETDEV WATCHDOG: CPU: 1: transmit queue 1 timed out 163860 ms
[115666.719772] mtk_soc_eth 1e100000.ethernet wan: NETDEV WATCHDOG: CPU: 1: transmit queue 1 timed out 168820 ms
[115666.739557] mtk_soc_eth 1e100000.ethernet eth0: NETDEV WATCHDOG: CPU: 1: transmit queue 4 timed out 310940 ms
[115671.759724] mtk_soc_eth 1e100000.ethernet eth0: NETDEV WATCHDOG: CPU: 1: transmit queue 4 timed out 315980 ms
[115671.779698] mtk_soc_eth 1e100000.ethernet wan: NETDEV WATCHDOG: CPU: 1: transmit queue 1 timed out 173860 ms
[115676.719659] mtk_soc_eth 1e100000.ethernet wan: NETDEV WATCHDOG: CPU: 1: transmit queue 1 timed out 178820 ms
[115676.739469] mtk_soc_eth 1e100000.ethernet eth0: NETDEV WATCHDOG: CPU: 1: transmit queue 4 timed out 320950 ms
[115681.759586] mtk_soc_eth 1e100000.ethernet eth0: NETDEV WATCHDOG: CPU: 1: transmit queue 4 timed out 325980 ms
[115681.779543] mtk_soc_eth 1e100000.ethernet wan: NETDEV WATCHDOG: CPU: 1: transmit queue 1 timed out 183870 ms
[115686.719532] mtk_soc_eth 1e100000.ethernet wan: NETDEV WATCHDOG: CPU: 1: transmit queue 1 timed out 188820 ms
[115686.739322] mtk_soc_eth 1e100000.ethernet eth0: NETDEV WATCHDOG: CPU: 1: transmit queue 4 timed out 330950 ms
[115691.759469] mtk_soc_eth 1e100000.ethernet eth0: NETDEV WATCHDOG: CPU: 1: transmit queue 4 timed out 335980 ms
[115691.779438] mtk_soc_eth 1e100000.ethernet wan: NETDEV WATCHDOG: CPU: 1: transmit queue 1 timed out 193880 ms
[115696.719411] mtk_soc_eth 1e100000.ethernet wan: NETDEV WATCHD+-----------------------------+med out 198820 ms
[115696.739201] mtk_soc_eth 1e100000.ethernet eth0: NETDEV WATCH|                             |imed out 340950 ms
[115701.759356] mtk_soc_eth 1e100000.ethernet eth0: NETDEV WATCH|  Cannot open /dev/ttyUSB0!  |imed out 345980 ms
[115701.779312] mtk_soc_eth 1e100000.ethernet wan: NETDEV WATCHD|                             |med out 203860 ms
[115706.719295] mtk_soc_eth 1e100000.ethernet wan: NETDEV WATCHD+-----------------------------+med out 208820 ms
[115706.739130] mtk_soc_eth 1e100000.ethernet eth0: NETDEV WATCHDOG: CPU: 1: transmit queue 4 timed out 350950 ms
[115711.759248] mtk_soc_eth 1e100000.ethernet eth0: NETDEV WATCHDOG: CPU: 1: transmit queue 4 timed out 355980 ms
[115711.779209] mtk_soc_eth 1e100000.ethernet wan: NETDEV WATCHDOG: CPU: 1: transmit queue 1 timed out 213860 ms
[115716.719166] mtk_soc_eth 1e100000.ethernet wan: NETDEV WATCHDOG: CPU: 1: transmit queue 1 timed out 218820 ms
[115716.738951] mtk_soc_eth 1e100000.ethernet eth0: NETDEV WATCHDOG: CPU: 1: transmit queue 4 timed out 360940 ms
[115721.759105] mtk_soc_eth 1e100000.ethernet eth0: NETDEV WATCHDOG: CPU: 1: transmit queue 4 timed out 365980 ms
[115721.779080] mtk_soc_eth 1e100000.ethernet wan: NETDEV WATCHDOG: CPU: 1: transmit queue 1 timed out 223860 ms
[115726.719049] mtk_soc_eth 1e100000.ethernet wan: NETDEV WATCHDOG: CPU: 1: transmit queue 1 timed out 228820 ms
[115726.738837] mtk_soc_eth 1e100000.ethernet eth0: NETDEV WATCHDOG: CPU: 1: transmit queue 4 timed out 370940 ms
[115731.758987] mtk_soc_eth 1e100000.ethernet eth0: NETDEV WATCHDOG: CPU: 1: transmit queue 4 timed out 375980 ms
[115731.778945] mtk_soc_eth 1e100000.ethernet wan: NETDEV WATCHDOG: CPU: 1: transmit queue 1 timed out 233860 ms
[115736.718933] mtk_soc_eth 1e100000.ethernet wan: NETDEV WATCHDOG: CPU: 1: transmit queue 1 timed out 238820 ms
[115736.738729] mtk_soc_eth 1e100000.ethernet eth0: NETDEV WATCHDOG: CPU: 1: transmit queue 4 timed out 380940 ms
[115741.758869] mtk_soc_eth 1e100000.ethernet eth0: NETDEV WATCHDOG: CPU: 1: transmit queue 4 timed out 385980 ms
[115741.778827] mtk_soc_eth 1e100000.ethernet wan: NETDEV WATCHDOG: CPU: 1: transmit queue 1 timed out 243860 ms
root@OpenWrt:~# [115751.758743] mtk_soc_eth 1e100000.ethernet eth0: NETDEV WATCHDOG: CPU: 1: transmit queue 3 timed out 7790 ms
[115751.778377] mtk_soc_eth 1e100000.ethernet wan: NETDEV WATCHDOG: CPU: 1: transmit queue 1 timed out 253870 ms
[115756.718680] mtk_soc_eth 1e100000.ethernet wan: NETDEV WATCHDOG: CPU: 1: transmit queue 1 timed out 258820 ms
[115756.738471] mtk_soc_eth 1e100000.ethernet eth0: NETDEV WATCHDOG: CPU: 1: transmit queue 3 timed out 12750 ms
[115761.758635] mtk_soc_eth 1e100000.ethernet eth0: NETDEV WATCHDOG: CPU: 1: transmit queue 3 timed out 17790 ms
[115761.778439] mtk_soc_eth 1e100000.ethernet wan: NETDEV WATCHDOG: CPU: 1: transmit queue 1 timed out 263870 ms
[115766.718558] mtk_soc_eth 1e100000.ethernet wan: NETDEV WATCHDOG: CPU: 1: transmit queue 1 timed out 268820 ms
[115766.738361] mtk_soc_eth 1e100000.ethernet eth0: NETDEV WATCHDOG: CPU: 1: transmit queue 3 timed out 22750 ms
[115771.758510] mtk_soc_eth 1e100000.ethernet eth0: NETDEV WATCHDOG: CPU: 1: transmit queue 3 timed out 27790 ms
[115771.778293] mtk_soc_eth 1e100000.ethernet wan: NETDEV WATCHDOG: CPU: 1: transmit queue 1 timed out 273860 ms

```

dangowrt commented:
@axet Please describe your setup in detail (router as well as connected devices) so I can try to reproduce the issue.

axet commented:
Same as before. My router keep crashing (without stacktrace). I tested a lot of patches in the #12143

My hardware:

The HLK-7621A module use the MT7621A chipset.
CPU: MT7621 / 880 MHz / MIPS
Flash: 32MB
RAM: 256MB

With master openwrt branch (aka SNAPSHOT branch). Issue has no pattern. Random crash / stop working. Used to print stack trace with "timeout" error message. Now it simply report timeouts.

dangowrt commented:
@axet Please also tell me what kind of devices you have connected to the Ethernet ports (10M/100M/1000M?), if hardware flow offloading is active and whether your configuration is a simple bridged AP or if are using a routed setup with PPPoE or DHCP.

axet commented:
All ports connected to Wifi AP stations and WAN to internet with static IP, one port to desktop. I tried to disable flow control using ethtool - nothing helped.

I do not think my issue related to pull this request. It just didn't help.

dangowrt commented:
@axet While it will definitely be more difficult to discover the root cause, at least the triggered reset of the Ethernet controller should have some effect -- it obviously doesn't right now, which is a bit weird.

vit9696 commented:
@dangowrt, try the scenario I posted in https://github.com/openwrt/openwrt/issues/13122#issuecomment-1883685990. It is reliable for me, i.e. I can experience a crash in 1-2 minutes or less. Also, perhaps the issue in question should be reopened?

dangowrt commented:
As the original bug report was for MT798x Filogic SoC family and there are reports the issue has been resolved for those, I think what is left is implementing proper reset also on MT7621, which is another problem.

RomanovDanila commented:
@dangowrt a question, should we use RX_DMA_L4_VALID instead of RX_DMA_L4_VALID_V2 after this patch:
https://github.com/openwrt/openwrt/blob/main/target/linux/generic/backport-6.6/751-02-STABLE-net-ethernet-mediatek-use-QDMA-instead-of-ADMAv2-on-.patch

source:
`[][HIGH][kernel][mt7986/81][eth][Change RX DMA L4 valid bit definition]`
https://git01.mediatek.com/plugins/gitiles/openwrt/feeds/mtk-openwrt-feeds/+/cae5a7c7b67340f714786e332e62004a54abb8e5

f00b4r0 commented:
@dangowrt can you backport this to 23.05 please? These supported devices are affected (and essentially unusable) there.

I just tested dropping these two patches into `pending-5.15` on 23.05 and I went from a few minutes uptime before crash to over 48h and counting now.

Thanks!

f00b4r0 commented:
spoke too soon, it crashed again after 3.5 days, but this time not on the first NETDEV WATCHDOG. There were subsequent timeouts but the device remained working "normally" until it panic'd. I've tried to extract the relevant bits from console, notice the timestamps:

```
[10308.116783] ------------[ cut here ]------------
[10308.131200] NETDEV WATCHDOG: eth0 (mtk_soc_eth): transmit queue 5 timed out
[10308.138169] WARNING: CPU: 2 PID: 0 at 0xffffffc0086c64b4
[10308.143466] Modules linked in: pppoe ppp_async option nft_fib_inet nf_flow_table_ipv6 nf_flow_table_ipv4 nf_flow_table_inet usb_wwan pptp pppox ppp_mppe ppp_generic nft_reject_ipv6 nft_reject_ipv4 nft_reject_inet nft_reject nft_redir nft_quota nft_objref nft_numgen nft_nat nft_masq nft_log nft_limit nft_hash nft_flow_offload nft_fib_ipv6 nft_fib_ipv4 nft_fib nft_ct nft_counter nft_chain_nat nf_tables nf_nat nf_flow_table nf_conntrack_netlink nf_conntrack mt7915e mt76_connac_lib mt76 mac80211 huawei_cdc_ncm cfg80211 cdc_ncm cdc_ether usbserial usbnet slhc nfnetlink nf_reject_ipv6 nf_reject_ipv4 nf_log_syslog nf_defrag_ipv6 nf_defrag_ipv4 libcrc32c compat cdc_wdm crypto_safexcel sch_tbf sch_ingress sch_htb sch_hfsc em_u32 cls_u32 cls_route cls_matchall cls_fw cls_flow cls_basic act_skbedit act_mirred act_gact ip_gre gre ifb ip_tunnel crypto_user algif_skcipher algif_rng algif_hash algif_aead af_alg sha1_generic seqiv md5 des_generic libdes authencesn authenc arc4 leds_gpio xhci_plat_hcd
[10308.143620]  xhci_pci xhci_mtk_hcd xhci_hcd gpio_button_hotplug usbcore usb_common aquantia crc_ccitt mii
[10308.240186] CPU: 2 PID: 0 Comm: swapper/2 Not tainted 5.15.167 #0
[10308.246263] Hardware name: GL.iNet GL-MT6000 (DT)
[10308.250949] pstate: 40400005 (nZcv daif +PAN -UAO -TCO -DIT -SSBS BTYPE=--)
[10308.257890] pc : 0xffffffc0086c64b4
[10308.261363] lr : 0xffffffc0086c64b4
[10308.264835] sp : ffffffc008b7bdb0
[10308.268133] x29: ffffffc008b7bdb0 x28: 0000000000000140 x27: 00000000ffffffff
[10308.275248] x26: 0000000000000000 x25: 0000000000000002 x24: ffffff8000e5e4c0
[10308.282364] x23: 0000000000000000 x22: 0000000000000001 x21: ffffffc008a46000
[10308.289480] x20: ffffff8000e5e000 x19: 0000000000000005 x18: ffffffc008a5a2a8
[10308.296595] x17: 393a6166206d6f72 x16: 66206e6f69746163 x15: 0000000000000402
[10308.303709] x14: 0000000000000156 x13: ffffffc008b7bae8 x12: ffffffc008ab22a8
[10308.310824] x11: 712074696d736e61 x10: ffffffc008ab22a8 x9 : 0000000000000000
[10308.317939] x8 : ffffffc008a5a258 x7 : ffffffc008a5a2a8 x6 : 0000000000000001
[10308.325055] x5 : 0000000000000000 x4 : 0000000000000000 x3 : 0000000000000000
[10308.332169] x2 : ffffff803fdb1090 x1 : ffffffc037383000 x0 : 000000000000003f
[10308.339285] Call trace:
[10308.341717]  0xffffffc0086c64b4
[10308.344843]  0xffffffc0080a5b50
[10308.347968]  0xffffffc0080a63c8
[10308.351093]  0xffffffc0080a6918
[10308.354218]  0xffffffc00803b04c
[10308.357343]  0xffffffc00803b404
[10308.360469]  0xffffffc00803b5d0
[10308.363594]  0xffffffc008089454
[10308.366719]  0xffffffc0080100e0
[10308.369845]  0xffffffc00801584c
[10308.372971]  0xffffffc00801609c
[10308.376095]  0xffffffc008836db0
[10308.379220]  0xffffffc008837564
[10308.382345]  0xffffffc008011374
[10308.385469]  0xffffffc008837b44
[10308.388594]  0xffffffc008069350
[10308.391719]  0xffffffc008069560
[10308.394844]  0xffffffc00801ebe0
[10308.397968]  0xffffffc0088402cc
[10308.401094] ---[ end trace 7850096a3aef9c52 ]---
[10308.405707] mtk_soc_eth 15100000.ethernet eth0: transmit timed out
[10308.544790] mtk_soc_eth 15100000.ethernet eth0: Link is Down
[10308.550472] mtk_soc_eth 15100000.ethernet eth1: Link is Down
[10309.216772] mtk_soc_eth 15100000.ethernet: warm reset failed
[10309.237482] mtk_soc_eth 15100000.ethernet eth0: configuring for fixed/2500base-x link mode
[10309.245903] mtk_soc_eth 15100000.ethernet eth0: Link is Up - 2.5Gbps/Full - flow control rx/tx
[10309.535403] mtk_soc_eth 15100000.ethernet eth1: PHY [mdio-bus:01] driver [RTL8221B-VB-CG 2.5Gbps PHY] (irq=62)
[10309.545650] mtk_soc_eth 15100000.ethernet eth1: configuring for phy/2500base-x link mode
[10309.586281] vbridge0: port 1(eth1) entered disabled state
[10313.457505] mtk_soc_eth 15100000.ethernet eth1: Link is Up - 2.5Gbps/Full - flow control rx/tx
[10313.457531] vbridge0: port 1(eth1) entered blocking state
[10313.471513] vbridge0: port 1(eth1) entered forwarding state
[...]
[99828.487950] mtk_soc_eth 15100000.ethernet eth1: transmit timed out
[99828.635103] mtk_soc_eth 15100000.ethernet eth0: Link is Down
[99828.640795] mtk_soc_eth 15100000.ethernet eth1: Link is Down
[99829.298174] mtk_soc_eth 15100000.ethernet: warm reset failed
[99829.319001] mtk_soc_eth 15100000.ethernet eth0: configuring for fixed/2500base-x link mode
[99829.327419] mtk_soc_eth 15100000.ethernet eth0: Link is Up - 2.5Gbps/Full - flow control rx/tx
[99829.636828] mtk_soc_eth 15100000.ethernet eth1: PHY [mdio-bus:01] driver [RTL8221B-VB-CG 2.5Gbps PHY] (irq=62)
[99829.647082] mtk_soc_eth 15100000.ethernet eth1: configuring for phy/2500base-x link mode
[99829.687756] vbridge0: port 1(eth1) entered disabled state
[99833.566766] mtk_soc_eth 15100000.ethernet eth1: Link is Up - 2.5Gbps/Full - flow control rx/tx
[99833.566795] vbridge0: port 1(eth1) entered blocking state
[99833.580767] vbridge0: port 1(eth1) entered forwarding state
[99889.452700] page_pool_release_retry() stalled pool shutdown 4 inflight 60 sec
[99949.708072] page_pool_release_retry() stalled pool shutdown 4 inflight 120 sec
[100009.965175] page_pool_release_retry() stalled pool shutdown 4 inflight 181 sec
[...]
[113379.053169] mtk_soc_eth 15100000.ethernet eth1: transmit timed out
[113379.201426] mtk_soc_eth 15100000.ethernet eth0: Link is Down
[113379.207197] mtk_soc_eth 15100000.ethernet eth1: Link is Down
[113379.873259] mtk_soc_eth 15100000.ethernet: warm reset failed
[113379.894205] mtk_soc_eth 15100000.ethernet eth0: configuring for fixed/2500base-x link mode
[113379.902707] mtk_soc_eth 15100000.ethernet eth0: Link is Up - 2.5Gbps/Full - flow control rx/tx
[113380.211883] mtk_soc_eth 15100000.ethernet eth1: PHY [mdio-bus:01] driver [RTL8221B-VB-CG 2.5Gbps PHY] (irq=62)
[113380.222240] mtk_soc_eth 15100000.ethernet eth1: configuring for phy/2500base-x link mode
[113380.262864] vbridge0: port 1(eth1) entered disabled state
[113384.268179] mtk_soc_eth 15100000.ethernet eth1: Link is Up - 2.5Gbps/Full - flow control rx/tx
[113384.268206] vbridge0: port 1(eth1) entered blocking state
[113384.282362] vbridge0: port 1(eth1) entered forwarding state
[...]
[116022.312548] mtk_soc_eth 15100000.ethernet eth0: transmit timed out
[116022.460830] mtk_soc_eth 15100000.ethernet eth0: Link is Down
[116022.466601] mtk_soc_eth 15100000.ethernet eth1: Link is Down
[116023.132648] mtk_soc_eth 15100000.ethernet: warm reset failed
[116023.153632] mtk_soc_eth 15100000.ethernet eth0: configuring for fixed/2500base-x link mode
[116023.162135] mtk_soc_eth 15100000.ethernet eth0: Link is Up - 2.5Gbps/Full - flow control rx/tx
[116023.451328] mtk_soc_eth 15100000.ethernet eth1: PHY [mdio-bus:01] driver [RTL8221B-VB-CG 2.5Gbps PHY] (irq=62)
[116023.461666] mtk_soc_eth 15100000.ethernet eth1: configuring for phy/2500base-x link mode
[116023.502265] vbridge0: port 1(eth1) entered disabled state
[116027.399228] mtk_soc_eth 15100000.ethernet eth1: Link is Up - 2.5Gbps/Full - flow control rx/tx
[116027.399255] vbridge0: port 1(eth1) entered blocking state
[116027.413408] vbridge0: port 1(eth1) entered forwarding state
[...]
[more timeouts like the above until:]
[322853.552640] mtk_soc_eth 15100000.ethernet eth1: transmit timed out
[322853.700908] mtk_soc_eth 15100000.ethernet eth0: Link is Down
[322853.706682] mtk_soc_eth 15100000.ethernet eth1: Link is Down
[322854.234903] Unable to handle kernel paging request at virtual address ffffffffd4780000
[322854.242931] Mem abort info:
[322854.245796]   ESR = 0x0000000096000145
[322854.249613]   EC = 0x25: DABT (current EL), IL = 32 bits
[322854.255004]   SET = 0, FnV = 0
[322854.258129]   EA = 0, S1PTW = 0
[322854.261341]   FSC = 0x05: level 1 translation fault
[322854.266296] Data abort info:
[322854.269249]   ISV = 0, ISS = 0x00000145
[322854.273164]   CM = 1, WnR = 1
[322854.276205] swapper pgtable: 4k pages, 39-bit VAs, pgdp=00000000489c8000
[322854.282980] [ffffffffd4780000] pgd=0000000000000000, p4d=0000000000000000, pud=0000000000000000
[322854.291751] Internal error: Oops: 0000000096000145 [#1] SMP
[322854.297392] Modules linked in: pppoe ppp_async option nft_fib_inet nf_flow_table_ipv6 nf_flow_table_ipv4 nf_flow_table_inet usb_wwan pptp pppox ppp_mppe ppp_generic nft_reject_ipv6 nft_reject_ipv4 nft_reject_inet nft_reject nft_redir nft_quota nft_objref nft_numgen nft_nat nft_masq nft_log nft_limit nft_hash nft_flow_offload nft_fib_ipv6 nft_fib_ipv4 nft_fib nft_ct nft_counter nft_chain_nat nf_tables nf_nat nf_flow_table nf_conntrack_netlink nf_conntrack mt7915e mt76_connac_lib mt76 mac80211 huawei_cdc_ncm cfg80211 cdc_ncm cdc_ether usbserial usbnet slhc nfnetlink nf_reject_ipv6 nf_reject_ipv4 nf_log_syslog nf_defrag_ipv6 nf_defrag_ipv4 libcrc32c compat cdc_wdm crypto_safexcel sch_tbf sch_ingress sch_htb sch_hfsc em_u32 cls_u32 cls_route cls_matchall cls_fw cls_flow cls_basic act_skbedit act_mirred act_gact ip_gre gre ifb ip_tunnel crypto_user algif_skcipher algif_rng algif_hash algif_aead af_alg sha1_generic seqiv md5 des_generic libdes authencesn authenc arc4 leds_gpio xhci_plat_hcd
[322854.297545]  xhci_pci xhci_mtk_hcd xhci_hcd gpio_button_hotplug usbcore usb_common aquantia crc_ccitt mii
[322854.394282] CPU: 2 PID: 20129 Comm: kworker/2:0 Tainted: G        W         5.15.167 #0
[322854.402351] Hardware name: GL.iNet GL-MT6000 (DT)
[322854.407124] Workqueue: events 0xffffffc0085d2284
[322854.411814] pstate: 40400005 (nZcv daif +PAN -UAO -TCO -DIT -SSBS BTYPE=--)
[322854.418841] pc : 0xffffffc008023114
[322854.422400] lr : 0xffffffc008022160
[322854.425958] sp : ffffffc009d3bc00
[322854.429342] x29: ffffffc009d3bc00 x28: 0000000000000000 x27: ffffff8000b3b980
[322854.436545] x26: 0000000000000008 x25: 0000000000000000 x24: 0000000014780000
[322854.443747] x23: 0000000000000800 x22: 0000000000000000 x21: 0000000000000002
[322854.450949] x20: ffffff8000099410 x19: 0000000014780000 x18: 0000000000000000
[322854.458151] x17: 0000000000000000 x16: 0000000000000000 x15: 0000000000000000
[322854.465353] x14: 0000000000000000 x13: 00005facb3f1d68d x12: 0000000000000006
[322854.472554] x11: ffffffc008a34d60 x10: 0000000000000000 x9 : ffffffc037383000
[322854.479756] x8 : 0000000000800000 x7 : 0000000000000000 x6 : 0000000000000800
[322854.486957] x5 : ffffffc0088c3018 x4 : 0000000000000000 x3 : 000000000000003f
[322854.494158] x2 : 0000000000000040 x1 : ffffffffd4780800 x0 : ffffffffd4780000
[322854.501361] Call trace:
[322854.503879]  0xffffffc008023114
[322854.507091]  0xffffffc0080a14dc
[322854.510302]  0xffffffc0085cc8f4
[322854.513513]  0xffffffc0085cf6a8
[322854.516724]  0xffffffc0085d1790
[322854.519935]  0xffffffc0085d22fc
[322854.523147]  0xffffffc0080508e0
[322854.526358]  0xffffffc0080516d0
[322854.529569]  0xffffffc008058e6c
[322854.532781]  0xffffffc00801581c
[322854.535996] Code: 8a230000 54000060 d50b7e20 14000002 (d5087620) 
[322854.542156] ---[ end trace 7850096a3aef9c53 ]---
[322854.564101] Kernel panic - not syncing: Oops: Fatal exception
[322854.569920] SMP: stopping secondary CPUs
[322854.573916] Kernel Offset: disabled
[322854.577474] CPU features: 0x0,00000000,20000802
[322854.582075] Memory Limit: none
[322854.602325] Rebooting in 3 seconds..
```

NovaPixell commented:
> spoke too soon, it crashed again after 3.5 days, but this time not on the first NETDEV WATCHDOG. There were subsequent timeouts but the device remained working "normally" until it panic'd. I've tried to extract the relevant bits from console, notice the timestamps:
> 
> ```
> [10308.116783] ------------[ cut here ]------------
> [10308.131200] NETDEV WATCHDOG: eth0 (mtk_soc_eth): transmit queue 5 timed out
> [10308.138169] WARNING: CPU: 2 PID: 0 at 0xffffffc0086c64b4
> [10308.143466] Modules linked in: pppoe ppp_async option nft_fib_inet nf_flow_table_ipv6 nf_flow_table_ipv4 nf_flow_table_inet usb_wwan pptp pppox ppp_mppe ppp_generic nft_reject_ipv6 nft_reject_ipv4 nft_reject_inet nft_reject nft_redir nft_quota nft_objref nft_numgen nft_nat nft_masq nft_log nft_limit nft_hash nft_flow_offload nft_fib_ipv6 nft_fib_ipv4 nft_fib nft_ct nft_counter nft_chain_nat nf_tables nf_nat nf_flow_table nf_conntrack_netlink nf_conntrack mt7915e mt76_connac_lib mt76 mac80211 huawei_cdc_ncm cfg80211 cdc_ncm cdc_ether usbserial usbnet slhc nfnetlink nf_reject_ipv6 nf_reject_ipv4 nf_log_syslog nf_defrag_ipv6 nf_defrag_ipv4 libcrc32c compat cdc_wdm crypto_safexcel sch_tbf sch_ingress sch_htb sch_hfsc em_u32 cls_u32 cls_route cls_matchall cls_fw cls_flow cls_basic act_skbedit act_mirred act_gact ip_gre gre ifb ip_tunnel crypto_user algif_skcipher algif_rng algif_hash algif_aead af_alg sha1_generic seqiv md5 des_generic libdes authencesn authenc arc4 leds_gpio xhci_plat_hcd
> [10308.143620]  xhci_pci xhci_mtk_hcd xhci_hcd gpio_button_hotplug usbcore usb_common aquantia crc_ccitt mii
> [10308.240186] CPU: 2 PID: 0 Comm: swapper/2 Not tainted 5.15.167 #0
> [10308.246263] Hardware name: GL.iNet GL-MT6000 (DT)
> [10308.250949] pstate: 40400005 (nZcv daif +PAN -UAO -TCO -DIT -SSBS BTYPE=--)
> [10308.257890] pc : 0xffffffc0086c64b4
> [10308.261363] lr : 0xffffffc0086c64b4
> [10308.264835] sp : ffffffc008b7bdb0
> [10308.268133] x29: ffffffc008b7bdb0 x28: 0000000000000140 x27: 00000000ffffffff
> [10308.275248] x26: 0000000000000000 x25: 0000000000000002 x24: ffffff8000e5e4c0
> [10308.282364] x23: 0000000000000000 x22: 0000000000000001 x21: ffffffc008a46000
> [10308.289480] x20: ffffff8000e5e000 x19: 0000000000000005 x18: ffffffc008a5a2a8
> [10308.296595] x17: 393a6166206d6f72 x16: 66206e6f69746163 x15: 0000000000000402
> [10308.303709] x14: 0000000000000156 x13: ffffffc008b7bae8 x12: ffffffc008ab22a8
> [10308.310824] x11: 712074696d736e61 x10: ffffffc008ab22a8 x9 : 0000000000000000
> [10308.317939] x8 : ffffffc008a5a258 x7 : ffffffc008a5a2a8 x6 : 0000000000000001
> [10308.325055] x5 : 0000000000000000 x4 : 0000000000000000 x3 : 0000000000000000
> [10308.332169] x2 : ffffff803fdb1090 x1 : ffffffc037383000 x0 : 000000000000003f
> [10308.339285] Call trace:
> [10308.341717]  0xffffffc0086c64b4
> [10308.344843]  0xffffffc0080a5b50
> [10308.347968]  0xffffffc0080a63c8
> [10308.351093]  0xffffffc0080a6918
> [10308.354218]  0xffffffc00803b04c
> [10308.357343]  0xffffffc00803b404
> [10308.360469]  0xffffffc00803b5d0
> [10308.363594]  0xffffffc008089454
> [10308.366719]  0xffffffc0080100e0
> [10308.369845]  0xffffffc00801584c
> [10308.372971]  0xffffffc00801609c
> [10308.376095]  0xffffffc008836db0
> [10308.379220]  0xffffffc008837564
> [10308.382345]  0xffffffc008011374
> [10308.385469]  0xffffffc008837b44
> [10308.388594]  0xffffffc008069350
> [10308.391719]  0xffffffc008069560
> [10308.394844]  0xffffffc00801ebe0
> [10308.397968]  0xffffffc0088402cc
> [10308.401094] ---[ end trace 7850096a3aef9c52 ]---
> [10308.405707] mtk_soc_eth 15100000.ethernet eth0: transmit timed out
> [10308.544790] mtk_soc_eth 15100000.ethernet eth0: Link is Down
> [10308.550472] mtk_soc_eth 15100000.ethernet eth1: Link is Down
> [10309.216772] mtk_soc_eth 15100000.ethernet: warm reset failed
> [10309.237482] mtk_soc_eth 15100000.ethernet eth0: configuring for fixed/2500base-x link mode
> [10309.245903] mtk_soc_eth 15100000.ethernet eth0: Link is Up - 2.5Gbps/Full - flow control rx/tx
> [10309.535403] mtk_soc_eth 15100000.ethernet eth1: PHY [mdio-bus:01] driver [RTL8221B-VB-CG 2.5Gbps PHY] (irq=62)
> [10309.545650] mtk_soc_eth 15100000.ethernet eth1: configuring for phy/2500base-x link mode
> [10309.586281] vbridge0: port 1(eth1) entered disabled state
> [10313.457505] mtk_soc_eth 15100000.ethernet eth1: Link is Up - 2.5Gbps/Full - flow control rx/tx
> [10313.457531] vbridge0: port 1(eth1) entered blocking state
> [10313.471513] vbridge0: port 1(eth1) entered forwarding state
> [...]
> [99828.487950] mtk_soc_eth 15100000.ethernet eth1: transmit timed out
> [99828.635103] mtk_soc_eth 15100000.ethernet eth0: Link is Down
> [99828.640795] mtk_soc_eth 15100000.ethernet eth1: Link is Down
> [99829.298174] mtk_soc_eth 15100000.ethernet: warm reset failed
> [99829.319001] mtk_soc_eth 15100000.ethernet eth0: configuring for fixed/2500base-x link mode
> [99829.327419] mtk_soc_eth 15100000.ethernet eth0: Link is Up - 2.5Gbps/Full - flow control rx/tx
> [99829.636828] mtk_soc_eth 15100000.ethernet eth1: PHY [mdio-bus:01] driver [RTL8221B-VB-CG 2.5Gbps PHY] (irq=62)
> [99829.647082] mtk_soc_eth 15100000.ethernet eth1: configuring for phy/2500base-x link mode
> [99829.687756] vbridge0: port 1(eth1) entered disabled state
> [99833.566766] mtk_soc_eth 15100000.ethernet eth1: Link is Up - 2.5Gbps/Full - flow control rx/tx
> [99833.566795] vbridge0: port 1(eth1) entered blocking state
> [99833.580767] vbridge0: port 1(eth1) entered forwarding state
> [99889.452700] page_pool_release_retry() stalled pool shutdown 4 inflight 60 sec
> [99949.708072] page_pool_release_retry() stalled pool shutdown 4 inflight 120 sec
> [100009.965175] page_pool_release_retry() stalled pool shutdown 4 inflight 181 sec
> [...]
> [113379.053169] mtk_soc_eth 15100000.ethernet eth1: transmit timed out
> [113379.201426] mtk_soc_eth 15100000.ethernet eth0: Link is Down
> [113379.207197] mtk_soc_eth 15100000.ethernet eth1: Link is Down
> [113379.873259] mtk_soc_eth 15100000.ethernet: warm reset failed
> [113379.894205] mtk_soc_eth 15100000.ethernet eth0: configuring for fixed/2500base-x link mode
> [113379.902707] mtk_soc_eth 15100000.ethernet eth0: Link is Up - 2.5Gbps/Full - flow control rx/tx
> [113380.211883] mtk_soc_eth 15100000.ethernet eth1: PHY [mdio-bus:01] driver [RTL8221B-VB-CG 2.5Gbps PHY] (irq=62)
> [113380.222240] mtk_soc_eth 15100000.ethernet eth1: configuring for phy/2500base-x link mode
> [113380.262864] vbridge0: port 1(eth1) entered disabled state
> [113384.268179] mtk_soc_eth 15100000.ethernet eth1: Link is Up - 2.5Gbps/Full - flow control rx/tx
> [113384.268206] vbridge0: port 1(eth1) entered blocking state
> [113384.282362] vbridge0: port 1(eth1) entered forwarding state
> [...]
> [116022.312548] mtk_soc_eth 15100000.ethernet eth0: transmit timed out
> [116022.460830] mtk_soc_eth 15100000.ethernet eth0: Link is Down
> [116022.466601] mtk_soc_eth 15100000.ethernet eth1: Link is Down
> [116023.132648] mtk_soc_eth 15100000.ethernet: warm reset failed
> [116023.153632] mtk_soc_eth 15100000.ethernet eth0: configuring for fixed/2500base-x link mode
> [116023.162135] mtk_soc_eth 15100000.ethernet eth0: Link is Up - 2.5Gbps/Full - flow control rx/tx
> [116023.451328] mtk_soc_eth 15100000.ethernet eth1: PHY [mdio-bus:01] driver [RTL8221B-VB-CG 2.5Gbps PHY] (irq=62)
> [116023.461666] mtk_soc_eth 15100000.ethernet eth1: configuring for phy/2500base-x link mode
> [116023.502265] vbridge0: port 1(eth1) entered disabled state
> [116027.399228] mtk_soc_eth 15100000.ethernet eth1: Link is Up - 2.5Gbps/Full - flow control rx/tx
> [116027.399255] vbridge0: port 1(eth1) entered blocking state
> [116027.413408] vbridge0: port 1(eth1) entered forwarding state
> [... more timeouts like the above until:]
> [322853.552640] mtk_soc_eth 15100000.ethernet eth1: transmit timed out
> [322853.700908] mtk_soc_eth 15100000.ethernet eth0: Link is Down
> [322853.706682] mtk_soc_eth 15100000.ethernet eth1: Link is Down
> [322854.234903] Unable to handle kernel paging request at virtual address ffffffffd4780000
> [322854.242931] Mem abort info:
> [322854.245796]   ESR = 0x0000000096000145
> [322854.249613]   EC = 0x25: DABT (current EL), IL = 32 bits
> [322854.255004]   SET = 0, FnV = 0
> [322854.258129]   EA = 0, S1PTW = 0
> [322854.261341]   FSC = 0x05: level 1 translation fault
> [322854.266296] Data abort info:
> [322854.269249]   ISV = 0, ISS = 0x00000145
> [322854.273164]   CM = 1, WnR = 1
> [322854.276205] swapper pgtable: 4k pages, 39-bit VAs, pgdp=00000000489c8000
> [322854.282980] [ffffffffd4780000] pgd=0000000000000000, p4d=0000000000000000, pud=0000000000000000
> [322854.291751] Internal error: Oops: 0000000096000145 [#1] SMP
> [322854.297392] Modules linked in: pppoe ppp_async option nft_fib_inet nf_flow_table_ipv6 nf_flow_table_ipv4 nf_flow_table_inet usb_wwan pptp pppox ppp_mppe ppp_generic nft_reject_ipv6 nft_reject_ipv4 nft_reject_inet nft_reject nft_redir nft_quota nft_objref nft_numgen nft_nat nft_masq nft_log nft_limit nft_hash nft_flow_offload nft_fib_ipv6 nft_fib_ipv4 nft_fib nft_ct nft_counter nft_chain_nat nf_tables nf_nat nf_flow_table nf_conntrack_netlink nf_conntrack mt7915e mt76_connac_lib mt76 mac80211 huawei_cdc_ncm cfg80211 cdc_ncm cdc_ether usbserial usbnet slhc nfnetlink nf_reject_ipv6 nf_reject_ipv4 nf_log_syslog nf_defrag_ipv6 nf_defrag_ipv4 libcrc32c compat cdc_wdm crypto_safexcel sch_tbf sch_ingress sch_htb sch_hfsc em_u32 cls_u32 cls_route cls_matchall cls_fw cls_flow cls_basic act_skbedit act_mirred act_gact ip_gre gre ifb ip_tunnel crypto_user algif_skcipher algif_rng algif_hash algif_aead af_alg sha1_generic seqiv md5 des_generic libdes authencesn authenc arc4 leds_gpio xhci_plat_hcd
> [322854.297545]  xhci_pci xhci_mtk_hcd xhci_hcd gpio_button_hotplug usbcore usb_common aquantia crc_ccitt mii
> [322854.394282] CPU: 2 PID: 20129 Comm: kworker/2:0 Tainted: G        W         5.15.167 #0
> [322854.402351] Hardware name: GL.iNet GL-MT6000 (DT)
> [322854.407124] Workqueue: events 0xffffffc0085d2284
> [322854.411814] pstate: 40400005 (nZcv daif +PAN -UAO -TCO -DIT -SSBS BTYPE=--)
> [322854.418841] pc : 0xffffffc008023114
> [322854.422400] lr : 0xffffffc008022160
> [322854.425958] sp : ffffffc009d3bc00
> [322854.429342] x29: ffffffc009d3bc00 x28: 0000000000000000 x27: ffffff8000b3b980
> [322854.436545] x26: 0000000000000008 x25: 0000000000000000 x24: 0000000014780000
> [322854.443747] x23: 0000000000000800 x22: 0000000000000000 x21: 0000000000000002
> [322854.450949] x20: ffffff8000099410 x19: 0000000014780000 x18: 0000000000000000
> [322854.458151] x17: 0000000000000000 x16: 0000000000000000 x15: 0000000000000000
> [322854.465353] x14: 0000000000000000 x13: 00005facb3f1d68d x12: 0000000000000006
> [322854.472554] x11: ffffffc008a34d60 x10: 0000000000000000 x9 : ffffffc037383000
> [322854.479756] x8 : 0000000000800000 x7 : 0000000000000000 x6 : 0000000000000800
> [322854.486957] x5 : ffffffc0088c3018 x4 : 0000000000000000 x3 : 000000000000003f
> [322854.494158] x2 : 0000000000000040 x1 : ffffffffd4780800 x0 : ffffffffd4780000
> [322854.501361] Call trace:
> [322854.503879]  0xffffffc008023114
> [322854.507091]  0xffffffc0080a14dc
> [322854.510302]  0xffffffc0085cc8f4
> [322854.513513]  0xffffffc0085cf6a8
> [322854.516724]  0xffffffc0085d1790
> [322854.519935]  0xffffffc0085d22fc
> [322854.523147]  0xffffffc0080508e0
> [322854.526358]  0xffffffc0080516d0
> [322854.529569]  0xffffffc008058e6c
> [322854.532781]  0xffffffc00801581c
> [322854.535996] Code: 8a230000 54000060 d50b7e20 14000002 (d5087620) 
> [322854.542156] ---[ end trace 7850096a3aef9c53 ]---
> [322854.564101] Kernel panic - not syncing: Oops: Fatal exception
> [322854.569920] SMP: stopping secondary CPUs
> [322854.573916] Kernel Offset: disabled
> [322854.577474] CPU features: 0x0,00000000,20000802
> [322854.582075] Memory Limit: none
> [322854.602325] Rebooting in 3 seconds..
> ```

Have you tried using the latest snapshot version? At the moment I am not experiencing any crashes on kenel 6.6

f00b4r0 commented:
> Have you tried using the latest snapshot version? At the moment I am not experiencing any crashes on kenel 6.6

The trouble is that the setup in which these devices are used cannot run from snapshots, which makes testing difficult.

Regardless, even if the issue is solved in master the point I'm trying to make here is that we claim these devices are "supported" in 23.05, but the reality is they are unusable there so IMHO we should either change that status or fix the problem.

yogo1212 commented:
@f00b4r0 hi :-)

they are "supported", only there seems to be a regress with mt7621.
for my the device in my home it has been working up until recently.
instead of "supported", there could be a "degraded" flag - but that's probably not really a good idea if we want the maintainers to have any time left for actually merging fixes.

i feel you, though.

dangowrt commented:
> Will openwrt create there own version of driver?

We intend to submit all patches upstream to Linux kernel, but some need a lot of time to discuss and in the meantime we can have them downstream.

> Should I create PR with mtk feeds fixes?

Yes, that would be great. Make sure to include patch descriptions and ideally also submit patches to upstream Linux.

> It's unlikely that mtk devs will upstream their patches.

Sadly... It would be much less work for everyone involved if they decided to just work on the upstream driver...
